tests/object_sync_test.go

```diff
@@ -1049,6 +1213,685 @@ func makeTestNotifications(t *testing.T) []Notification {
+		INNER JOIN dependency_edge tn ON tn.to_node_id = parent.id
+		INNER JOIN dependency_node child ON child.id = tn.from_node_id
+		INNER JOIN host child_host ON child_host.id = child.host_id
+		INNER JOIN redundancy_group_state rgs ON rgs.redundancy_group_id = rg.id
+		WHERE child_host.name IN (?)`,
+		totalExpectedChildren,
+	)
+	require.NoError(t, err, "expanding SQL IN clause for redundancy groups query")
+
+	var redundancyGroups []DependencyGroup
+	err = db.Select(&redundancyGroups, db.Rebind(query+` AND rg.display_name = ? GROUP BY rg.id, rg.display_name`), append(args, g.Name)...)
+	require.NoError(t, err, "fetching redundancy groups")
+
+	var redundancyGroup DependencyGroup
+	if g.IsRedundancyGroup() {
+		require.Lenf(t, redundancyGroups, 1, "there should be exactly one redundancy group %q", g.Name)
+		redundancyGroup = redundancyGroups[0]
+		g.Id = redundancyGroup.Id
+		assert.Equal(t, g.Name, redundancyGroup.Name, "redundancy group name should match")
+	} else {
+		assert.Lenf(t, redundancyGroups, 0, "there should be no redundancy group %q", g.Name)
+	}
+
+	type Checkable struct {
+		NodeId      types.Binary `db:""`
+		EdgeStateId types.Binary `db:"-"`
+		Id          types.Binary `db:"id"`
+		Name        string       `db:"name"`
+	}
+
+	// Perform some basic sanity checks on the hosts and redundancy groups (if any).
+	query, args, err = sqlx.In(
+		"SELECT id, name FROM host WHERE name IN (?)",
+		append(append([]string(nil), totalExpectedParents...), totalExpectedChildren...),
+	)
+	require.NoError(t, err, "expanding SQL IN clause for hosts query")
+
+	hostRows, err := db.Queryx(db.Rebind(query), args...)
+	require.NoError(t, err, "querying parent and child hosts")
+	defer hostRows.Close()
+
+	checkables := make(map[string]*Checkable)
+	for hostRows.Next() {
+		var c Checkable
+		require.NoError(t, hostRows.StructScan(&c), "scanning host row")
+		checkables[c.Name] = &c
+
+		// Retrieve the dependency node and dependency edge state ID of the current host.
+		assert.NoError(t, db.Get(&c.NodeId, db.Rebind(`SELECT id FROM dependency_node WHERE host_id = ?`), c.Id))
+		assert.NotNilf(t, c.NodeId, "host %q should have a dependency node", c.Name)
+
+		if slices.Contains(totalExpectedChildren, c.Name) && g.IsRedundancyGroup() {
+			assert.NoError(t, db.Get(&c.EdgeStateId, db.Rebind(`SELECT dependency_edge_state_id FROM dependency_edge WHERE to_node_id = ?`), redundancyGroup.Id))
+			assert.NotNilf(t, c.EdgeStateId, "host %q should have a dependency edge state", c.Name)
+		}
+	}
+	assert.NoError(t, hostRows.Err(), "scanned host rows should not have errors")
+	assert.Len(t, checkables, len(totalExpectedParents)+len(totalExpectedChildren), "all hosts should be in the database")
+
+	type Node struct {
+		RedundancyGroupId types.Binary `db:"node_id"`
+		Name              string       `db:"name"`
+		FromNodeId        types.Binary `db:"from_node_id"`
+		ToNodeId          types.Binary `db:"to_node_id"`
+	}
```

Comment:
Accompanying the change to the following query as mentioned in my previous comment, I'd also rename this type to `Edge` and the first two fields to `ParentRedundencyGroupId` and `ParentName`. For an edge, it's clear what the parent is in contrast to what `FromNodeId`/`ToNodeId` represents for a `Node` (as there may be multiple edges to/from that node whereas an edge has exactly one parent).

Bonus: this also fixes that strange-looking variable definition:
https://github.com/Icinga/icingadb/blob/2e806bfa4ebc0ee607d6f296245829d3e3ad3b87/tests/object_sync_test.go#L1588